test/visual_regression/services/visual_testing/take_percy_snapshot.js

```diff
@@ -20,18 +20,44 @@
 import { readFileSync } from 'fs';
 import { agentJsFilename } from '@percy/agent/dist/utils/sdk-utils';
 
-export function takePercySnapshot() {
+export function takePercySnapshot(show, hide) {
   if (!window.PercyAgent) {
     return false;
   }
 
+  const queryAll = selector => [
+    ...document.querySelectorAll(selector)
+  ];
+
+  const setSelectorClass = (selectors, className, remove) => {
+    for (const selector of selectors) {
+      for (const element of queryAll(selector)) {
+        if (remove) {
+          element.classList.remove(className);
+        } else {
+          console.log(className, element);
+
+          element.classList.add(className);
+        }
+      }
+    }
+  };
+
+  const hideByDefault = show.length > 0;
   const agent = new window.PercyAgent({
     handleAgentCommunication: false
   });
 
-  const queryAll = selector => [
-    ...document.querySelectorAll(selector)
-  ];
+  if (hideByDefault) {
+    document.body.classList.add('percyHideAllByDefault');
+  } else {
+    document.body.classList.remove('percyHideAllByDefault');
+  }
+
+  // set Percy visibility on elements
+  const className = hideByDefault ? 'showInPercy' : 'hideInPercy';
+  const selectors = hideByDefault ? show : hide;
+  setSelectorClass(selectors, className, false);
```

Comment:
We should probably apply `showInPercy` and `hideInPercy` to all elements regardless of `hideByDefault`, since we might want to hide an element that's within a show selector.

Reply:
Ok if you want this ability I would have to add `hideInPercy` to all dom elements to avoid using `*` makes hiding and showing children impossible. Any thoughts?

```scss
body.percyHideAllByDefault {
  * {
    visibility: hidden;
  }

  .hideInPercy * {
    visibility: hidden;
  }

  .showInPercy * {
    visibility: visible;
  }
}
```

Reply:
Eh, if it's not easy and it's not necessary right now then I don't mind skipping it. I thought it would be easy but nm!